### PR TITLE
Feature: Menu with core functionalities

### PR DESCRIPTION
**Pull request by usulpro**
#### step towards: #124 #868 #862

At this moment we can only use keyboard shortcuts and query string to get core functionalities. In some cases, we'd like to do the same via mouse or touch screen. Especially necessary for access from **mobile devices**

#### 1. Added menu duplicate keyboard shortcuts to the left (stories) panel

<details>
<summary>
expanded menu

</summary>

image

</details>

<details>
<summary>
  This menu is collapsible (default state):

</summary>

image

</details>

<details>
<summary>
  This menu has buttons to navigate within stories / storyKinds

</summary>

image

We'll need it in `Fullscreen` mode

</details>

<details>
<summary>
 Removed `⌘` button from the header of left panel

</summary>

image

It's still available from the core menu

</details>

#### 2. Added floating menu when left panel is hidden

<details>
<summary>
 Fullscreen mode

</summary>

image

So we can navigate via mouse/touchscreen or return to the normal view

</details>

<details>
<summary>
 Hidden left panel

</summary>

image

</details>

<details>
<summary>
 Expanse direction

</summary>

image

Depending on its position menu expands to up or down direction

</details>

#### 3. Mantra modules API

- `jumpToKind` - new API to jump between storyKinds: [modules/api/actions](https://github.com/UsulPro/storybook/blob/core_func_menu/packages/storybook-ui/src/modules/api/actions/api.js#L95)

- `floatingBoxOptions` - settings of floating menu displaying. [Default](https://github.com/UsulPro/storybook/blob/core_func_menu/packages/storybook-ui/src/modules/ui/index.js#L13):

```js
floatingBoxOptions: { 
      showFloatingBox: true, 
      floatingBoxPosition: boxPositions.BOTTOM_LEFT, 
}
```

### How to test

- Open [demo page](https://usulpro.github.io/storybook-ui/) to test usability. Try it with mobile devices.

- Clone repo
```shell
cd packages/storybook-ui
npm link
cd sample_repo
npm link @kadira/storybook-ui
npm start storybook
```

Try to use new menu to hide/show left/down panels. Go to `fullscreen`/`normal` mode. Navigate through the stories.

To change floating menu position set [default settings](https://github.com/UsulPro/storybook/blob/core_func_menu/packages/storybook-ui/src/modules/ui/index.js#L13) to one of [this](https://github.com/UsulPro/storybook/blob/core_func_menu/packages/storybook-ui/src/libs/menu_positions.js)

- Run tests

```shell
cd storybook
npm run test
```

All tests should pass

<details>
<summary> Details for review

</summary>

- New react component for menu is here: `packages/storybook-ui/src/modules/ui/components/core_menu/`

- It gets access to API via `react-komposer` here: `packages/storybook-ui/src/modules/ui/containers/core_menu.js`

-  new `jumpToKind` API is here: `packages/storybook-ui/src/modules/api/actions/api.js`

- all `svg` images for menu in the webpack bundled library: `packages/storybook-ui/src/modules/ui/components/assets/svg_package.js`. we can discuss this solution

</details>

### CODE REVIEW

**Comment by lab80** [comment]
Awesome!! Great feature!!

**Comment by danielduan** [comment]
I would love to see this merged sooner rather than later. Can you rebase this?

File: packages/storybook-ui/src/modules/ui/components/assets/svg_package.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
+/* eslint-disable */
+(function webpackUniversalModuleDefinition(root, factory) {
```
Comment (danielduan) [inline packages/storybook-ui/src/modules/ui/components/assets/svg_package.js]: I'm not following why we're committing a webpack generated module into the repo with svgs in base64 string format.

I think for maintainability sake, we should include the original svgs in the repo and generate them during runtime.

Comment (UsulPro) [inline packages/storybook-ui/src/modules/ui/components/assets/svg_package.js]: Thank you @danielduan 
It's a legacy solution from PR for Storybook 2.0 :)
I think we can switch to the "normal" way to import svg.
We'll need to change some loaders in webpack config, for that. Actually, I think we need to do it very carefully!

**Comment by danielduan** [comment]
Closing this because there hasn't been any activity here for a while. Feel free to reopen when this is ready for discussion again.

**Comment by bjankord** [comment]
I'm really interested in this PR, the ability to toggle all the panels makes viewing stories a lot easier on mobile devices. I'm currently writing scripts in manager-head.html to toggle the panels but its very much hacked together, I'd much prefer a first class storybook solution to handle this. @UsulPro, you mentioned needing to change some loaders in webpack config. Is that work you might be interested in picking up?

**Comment by danielduan** [comment]
@bjankord he's been unresponsive for a while now. The best thing you can do is pick up where he left off and try working on it.

**Pull request closed without merging**